### PR TITLE
AC-409 improve of views

### PR DESCRIPTION
**Pull request by davidguajardoo**
## Description ##
- UseCase updated with generic function for viewData class.
- Changes in viewModel.
- Datepicket, calendar and format dates optimized.
 
## Type of Change ##
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Unit or UI test
## Modules Branches Image ##
image

## Linked JIRA task or history ##
- https://carsales.atlassian.net/browse/AC-409
- https://carsales.atlassian.net/browse/AC-410

### CODE REVIEW

File: app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt [hunk 1]
```diff
@@ -24,88 +21,45 @@ class MainActivity : AppCompatActivity() {
...
-                is Resource.Error -> {
-                    hideLoading()
-                    it.message?.let {
-                        Toast.makeText(this, "An error ocurred : $it", Toast.LENGTH_LONG).show()
-                    }
-                }
-                else -> {
+    private fun initCalendar() {
+        CalendarManager.showPickerDialog(this) {
+            mDate = it
```
Comment (aapavezcs) [inline app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt]: Do you need mDate? 🤔

File: app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt [hunk 2]
```diff
@@ -24,88 +21,45 @@ class MainActivity : AppCompatActivity() {
...
+            displayFormatDate(it)
+            viewModel.getStatusCovid(mDate.toDateFormat(YYYY_MM_DD_WITH_SCRIPT))
+        }
+    }
 
+    private fun initObserver() {
+        viewModel.statusCovidStateLiveData.observe(this) {
+            if (it is State.Error) {
+                mBinding.errorView.show("An error ocurred : ${it.message}") {
```
Comment (aapavezcs) [inline app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt]: In this case,  it is better to use String.xml for any text in your fragment or activity.

**Review by aapavezcs — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt (hunk 1 above)
Comment (davidguajardoo) [reply app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt]: no. mDate removed

File: app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt (hunk 2 above)
Comment (davidguajardoo) [reply app/src/main/java/com/david/statuscovid_carsales/presentation/MainActivity.kt]: String resource is created.

**Review by aapavezcs — APPROVED** [review]

**Pull request merged**